# reason:
#   [110/327] Generating codebookjvm.c
#   FAILED: src/codebookjvm.c /builddir/codec2-1.0.1/build/src/codebookjvm.c
#   /bin/sh: line 1: /builddir/codec2-1.0.1/build/src/generate_codebook: cannot execute binary file: Exec format error

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -68,6 +68,7 @@
     ExternalProject_Add(codec2_native
        SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/..
        BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/codec2_native
+       CMAKE_ARGS -DCMAKE_C_COMPILER=$ENV{CC_FOR_BUILD} -DCMAKE_C_FLAGS=$ENV{CFLAGS_FOR_BUILD}
        BUILD_COMMAND ${CMAKE_COMMAND} --build . --target generate_codebook
        INSTALL_COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/codec2_native/src/generate_codebook ${CMAKE_CURRENT_BINARY_DIR}
        BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/generate_codebook
